core/src/test/scala/org/apache/spark/ui/UISuite.scala

```diff
@@ -190,6 +193,36 @@ class UISuite extends SparkFunSuite {
     }
   }
 
+  test("verify proxy rewrittenURI") {
```

Comment:
Could you add a test for encoded string in the url, such as `%2F`, `%F0%9F%98%84` (the 😄 emoji) ?